File: react-demo/src/main/java/com/example/reactdemo/service/EmployeeService.java
package com.example.reactdemo.service;

import java.util.List;

import com.example.reactdemo.models.Employee;

public interface EmployeeService {

    List<Employee> getAllEmployees();

    void addEmployee(final Employee employee);

    void updateEmployee(final Employee employee);

    void deleteEmployee(final Integer id);

    Employee getEmployeeById(final Integer id);
}

File: employee-react-app/src/service/EmployeeService.js
import axios from 'axios';

const EMPLOYEE_BASE_URL = "http://localhost:8081/employee"

class EmployeeService {

    getAllEmployees() {
        return axios.get(EMPLOYEE_BASE_URL);
    }

    addEmployee(employee) {
        return axios.post(EMPLOYEE_BASE_URL + "/create", employee);
    }

    getEmployeeById(id) {
        return axios.get(EMPLOYEE_BASE_URL + "/view/" + id);
    }

    updateEmployee(employee) {
        return axios.put(EMPLOYEE_BASE_URL + "/update", employee);
    }

    deleteEmployee(id) {
        return axios.delete(EMPLOYEE_BASE_URL + "/delete/" + id);
    }
}

export default new EmployeeService();
